lib/web_ui/test/engine/host_node_test.dart

```diff
@@ -33,30 +33,90 @@ void testMain() {
     });
 
     test('Attaches a stylesheet to the shadow root', () {
-      final DomElement firstChild =
-          (hostNode.node as DomShadowRoot).childNodes.toList()[0] as DomElement;
+      final DomElement? style =
+          hostNode.querySelector('#flt-internals-stylesheet');
 
-      expect(firstChild.tagName, equalsIgnoringCase('style'));
+      expect(style, isNotNull);
+      expect(style!.tagName, equalsIgnoringCase('style'));
+    });
+
+    test('(Self-test) hasCssRule can extract rules', () {
+      final DomElement? style =
+          hostNode.querySelector('#flt-internals-stylesheet');
+
+      final bool hasRule = hasCssRule(style,
+          selector: '.flt-text-editing::placeholder',
+          declaration: 'opacity: 0');
+
+      final bool hasFakeRule = hasCssRule(style,
+          selector: 'input::selection', declaration: 'color: #fabada;');
+
+      expect(hasRule, isTrue);
+      expect(hasFakeRule, isFalse);
+    });
+
+    test('Attaches outrageous text styles to flt-scene-host', () {
+      final DomElement? style =
+          hostNode.querySelector('#flt-internals-stylesheet');
+
+      final bool hasColorRed = hasCssRule(style,
+          selector: 'flt-scene-host', declaration: 'color: red');
+
+      bool hasFont = false;
+      if (isSafari) {
+        // Safari expands the shorthand rules, so we check for all we've set (separately).
+        hasFont = hasCssRule(style,
+                selector: 'flt-scene-host',
+                declaration: 'font-family: monospace') &&
+            hasCssRule(style,
+                selector: 'flt-scene-host', declaration: 'font-size: 14px');
+      } else {
+        hasFont = hasCssRule(style,
+            selector: 'flt-scene-host', declaration: 'font: 14px monospace');
+      }
+
+      expect(hasColorRed, isTrue,
+          reason: 'Should make foreground color red within scene host.');
+      expect(hasFont, isTrue, reason: 'Should pass default css font.');
     });
 
     test('Attaches styling to remove password reveal icons on Edge', () {
-      final DomElement? edgeStyleElement = hostNode.querySelector('#ms-reveal');
+      final DomElement? style =
+          hostNode.querySelector('#flt-internals-stylesheet');
 
-      expect(edgeStyleElement, isNotNull);
-      expect(edgeStyleElement!.innerText, 'input::-ms-reveal {display: none;}');
+      // Check that style.sheet! contains input::-ms-reveal rule
+      final bool hidesRevealIcons = hasCssRule(style,
+          selector: 'input::-ms-reveal', declaration: 'display: none');
+
+      final bool codeRanInFakeyBrowser = hasCssRule(style,
+          selector: 'input.fallback-for-fakey-browser-in-ci',
+          declaration: 'display: none');
+
+      if (codeRanInFakeyBrowser) {
+        print('Please, fix https://github.com/flutter/flutter/issues/116302');
```

Comment:
Nit: I think this issue relates to running Edge tests locally, not necessarily adding Edge support to our CI tests.  Should I create a new issue to track that?

Reply:
Yes please, if you feel that this needs a new issue, do it. If you end up creating a new issue, let me know, and I'll update the error message :)